Formula/vcpkg.rb

```diff
@@ -0,0 +1,29 @@
+class Vcpkg < Formula
+  desc "C++ Library Manager for Windows, Linux, and MacOS"
+  homepage "https://docs.microsoft.com/en-us/cpp/vcpkg"
+
+  stable do
+    url "https://github.com/microsoft/vcpkg/archive/2019.09.tar.gz"
+    sha256 "16c1b151cc3cf2022b03b699970e548f48cd9cd1e68db0b9c23a1d0d39391e56"
+  end
+
+  head do
+    url "https://github.com/microsoft/vcpkg.git"
+  end
+
+  depends_on "cmake" => :build
+  depends_on "ninja" => :build
+  depends_on "gcc"
+  
+  def install
+    ENV["CC"] = Formula["gcc"].opt_bin/"gcc-#{Formula["gcc"].version_suffix}"
+    ENV["CXX"] = Formula["gcc"].opt_bin/"g++-#{Formula["gcc"].version_suffix}"
+    system "./bootstrap-vcpkg.sh", "-useSystemBinaries"
+    libexec.install ["ports", "scripts", "triplets", "vcpkg", ".vcpkg-root"]
+    bin.install_symlink libexec/"vcpkg"
+  end
+
+  test do
+    assert_match version.to_s, pipe_output("#{bin}/vcpkg version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I have added a test for the  `search` command.

Do we need more test for the `install` command? It need download some files.

Reply:
@SMillerDev 
What else needs to be done?